### PR TITLE
Toml instead of Properties

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Moved `fs` to `commonMain` source set of `save-cloud-common` as `expected val` with `actual`s on all the platforms
 * Added `nativeMain` source set to `save-cloud-common`
 * Deleted `.properties` parsing logic
 * Moved from `Properties` configuration to `Toml` configuration (for agents)

### CODE REVIEW

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-cloud-common/src/jsMain/kotlin/com/saveourtool/save/utils/FileUtils.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * js implementation for FileUtils - Not implemented
+ */
+
+package com.saveourtool.save.utils
+
+import okio.FileSystem
+import okio.Path
+
+const val NOT_IMPLEMENTED_ON_JS = "Cannot be used in js."
```
Comment (nulls) [inline save-cloud-common/src/jsMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:10]: it cannot be used in legacy JS

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/FileUtils.kt [hunk 2]
```diff
@@ -97,12 +107,16 @@ fun Path.pathNamesTill(stop: Path): List<String> = generateSequence(this, Path::
...
-                "The path is not absolute: $this"
-            }
-        }
+fun Path.requireIsAbsolute(): Path = apply {
+    require(isAbsolute) {
+        "The path is not absolute: $this"
+    }
+}
+
+actual inline fun <reified C : Any> parseConfig(configPath: okio.Path): C {
```
Comment (nulls) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:116]: maybe move to `commonNonJsMain`?

File: save-cloud-common/build.gradle.kts [hunk 3]
```diff
@@ -81,6 +87,17 @@ kotlin {
                 runtimeOnly(libs.junit.jupiter.engine)
             }
         }
+
+        val linuxX64Main by getting
+        val macosX64Main by getting
+        val mingwX64Main by getting
+
+        val nativeMain by creating {
```
Comment (nulls) [inline save-cloud-common/build.gradle.kts:95]: why do we need `nativeMain` here?

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/FileUtils.kt (hunk 2 above)
Comment (nulls) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:116]: discussed offline, that it doesn't resolve the code duplication for different targets

File: save-cloud-common/src/jsMain/kotlin/com/saveourtool/save/utils/FileUtils.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-cloud-common/src/jsMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:10]: It cannot be used in js as well as js legacy, can it?

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/FileUtils.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:116]: Discussed offline - decided to leave as is

File: save-cloud-common/build.gradle.kts (hunk 3 above)
Comment (sanyavertolet) [reply save-cloud-common/build.gradle.kts:95]: There is an implementation for `linuxX64`, `macosX64` and `mingwX64`

Comment (nulls) [reply save-cloud-common/build.gradle.kts:95]: discussed offline: gradle has problem with long hierarchy

File: save-cloud-common/src/jsMain/kotlin/com/saveourtool/save/utils/FileUtils.kt (hunk 1 above)
Comment (nulls) [reply save-cloud-common/src/jsMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:10]: yes, you're right. I confused with serialization

**Review by nulls — APPROVED** [review]

**Pull request merged**